marklogic-client-api/src/main/java/com/marklogic/client/dataservices/impl/ExecEndpointImpl.java

```diff
@@ -73,28 +76,37 @@ public BulkExecCaller bulkCaller(CallContext callContext) {
 
     @Override
     public BulkExecCaller bulkCaller(CallContext[] callContexts) {
-        return null;
+        if(callContexts == null || callContexts.length==0)
+            throw new IllegalArgumentException("CallContext cannot be null or empty");
+        return bulkCaller(callContexts, callContexts.length);
```

Comment:
This review doesn't repeat the comments from the review of BulkExecCallerImpl earlier today.

This review assumes that the comments from the earlier review that apply to the other bulk callers don't have to be repeated on those bulk callers.  For one example, the comment on having a single thread count field in the base BulkIOCallerImpl applies generally.  For another example, the feedback on processInput() and on submitting the callable after putting the call context is similar to the comments on processOutput() and the callable in BulkExecCallerImpl.